### PR TITLE
HTML5 support

### PR DESCRIPTION
**Pull request by AGulev**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,31 +12,45 @@ Or point to the ZIP file of a [specific release](https://github.com/britzl/defol
...
+	-- On HTM5 screenshot will return as base64 image in a callback
+	local function html5_screenshot_callback(self, base64_img)
+	-- share your base64 png
+	end
+
+	-- Do not forget to check is this an html5 platform
+	if screenshot.html5 then
+		screenshot.html5(html5_screenshot_callback)
+	else
+		-- these functions do not available on html5
```
Comment (britzl) [inline README.md]: do not available -> are not available

File: screenshot/include/screenshot.h [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+#if defined(DM_PLATFORM_HTML5)
+
+typedef void (*JsCallback)(const char* base64image);
+
+extern "C" {
+    void screenshot_on_the_next_frame(JsCallback callback);
+}
+
+struct SreenshotLuaListener {
```
Comment (britzl) [inline screenshot/include/screenshot.h]: SreenshotLuaListener -> ScreenshotLuaListener

**Comment by AGulev** [comment]
Thank you!

File: README.md (hunk 1 above)
Comment (AGulev) [reply README.md]: Fixed, thx!

File: screenshot/include/screenshot.h [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+}
+
+struct ScreenshotLuaListener {
+ScreenshotLuaListener() : m_L(0), m_Callback(LUA_NOREF), m_Self(LUA_NOREF) {}
+	lua_State* m_L;
+	int m_Callback;
+	int m_Self;
+};
+
+void UnregisterCallback(lua_State* L);
```
Comment (britzl) [inline screenshot/include/screenshot.h]: This should be static to limit the scope to the current file.

File: screenshot/src/screenshot.cpp [hunk 4]
```diff
@@ -175,13 +174,80 @@ static int Buffer(lua_State* L) {
 	// Return 3 items
 	return 3;
 }
+#else
+ScreenshotLuaListener cbk;
+
+void UnregisterCallback(lua_State* L)
```
Comment (britzl) [inline screenshot/src/screenshot.cpp]: static

**Comment by britzl** [comment]
Thanks!

**Pull request merged**